synapse/lib/stormlib/model.py

```diff
@@ -919,8 +919,74 @@ class LibModelMigrations(s_stormtypes.Lib, MigrationEditorMixin):
                        'desc': 'Do not copy nodedata to the risk:vulnerable node.'},
                  ),
                  'returns': {'type': 'list', 'desc': 'A list of idens for the risk:vulnerable nodes.'}}},
+        {'name': 'inetSslCertToTlsServerCert', 'desc': '''
+            Create a inet:tls:servercert node from the provided inet:ssl:cert node.
+
+            Edits will be made to the inet:tls:servercert node in the current write layer.
+
+            Tags, tag properties, edges, and node data will be copied
+            to the inet:tls:servercert node. However, existing tag properties and
+            node data will not be overwritten.
+        ''',
+        'type': {'type': 'function', '_funcname': '_storm_query',
+                 'args': (
+                      {'name': 'n', 'type': 'node', 'desc': 'The inet:ssl:cert node to migrate.'},
+                      {'name': 'nodata', 'type': 'bool', 'default': False,
+                       'desc': 'Do not copy nodedata to the inet:tls:servercert node.'},
+                 ),
+                 'returns': {'type': 'list', 'desc': 'A list of idens for the inet:tls:servercert nodes.'}}},
+
     )
     _storm_lib_path = ('model', 'migration', 's')
+    _storm_query = '''
+        function _getNode(form, valu) {
+            *$form = $valu
+            return($node)
+        }
+
+        function _getCryptoNode(file) {
+            if (not $file) {
+                return()
+            }
+
+            crypto:x509:cert:file = $file
+            return($node)
+
+            $filenode = $_getNode(file:bytes, $file)
+            if ($filenode and $filenode.props.sha256) {
+                crypto:x509:cert:sha256 = $filenode.props.sha256
+                return($node)
+            }
+
+            [ crypto:x509:cert=($file,)
+                :file?=$file
```

Comment:
It might work best to make the main argument be `$sha256` that way you can populate both here. Any non-notional files should have a `:sha256` property :+1:

Reply:
I reworked that whole function. I think it's a lot better now.